test/test_nlp.jl

```diff
@@ -372,6 +372,40 @@ function test_parse_prod()
     return
 end
 
+function test_parse_minimum()
+    model = Model()
+    @variable(model, x[1:2])
+    ex = @NLexpression(model, minimum(x[i] for i in 1:2))
+    @test sprint(show, ex) == "subexpression[1]: min(x[1], x[2])"
+    ex2 = @NLexpression(model, minimum(x[i] for i in 1:0))
+    @test sprint(show, ex2) == "subexpression[2]: Inf"
```

Comment:
I'm open to other suggestions for what we should return here instead of `Inf`. Not sure whether it's better to throw an error here instead?

Reply:
Thinking on this, I don't think there's ever a case in which falling back to `Inf` can lead to a good model. It's always a modeling error, so I've changed to throwing a similar error to the one you get if you call `minimum(x for x in 1:0)`.